File: wordladder/wordladder_test.go
package wordladder_test

import (
	"testing"

	"github.com/hsmtkk/fuzzy-giggle/wordladder"
	"github.com/stretchr/testify/assert"
)

func Test1(t *testing.T) {
	words := wordladder.NewWords([]string{"apPle", "error", "suBway", "Zb"})
	problem := wordladder.NewProblem(words)
	chained, got := wordladder.New().Solve(problem)
	want := wordladder.NewAnswer(wordladder.NewChar("r"), wordladder.NewChar("s"))
	assert.False(t, chained)
	assert.Equal(t, want, got)
}

func Test2(t *testing.T) {
	words := wordladder.NewWords([]string{"idIOh", "hiKoQA", "AbijoD", "djgeldi", "kjoial"})
	problem := wordladder.NewProblem(words)
	chained, got := wordladder.New().Solve(problem)
	want := wordladder.NewAnswer(wordladder.NewChar("D"), wordladder.NewChar("d"))
	assert.False(t, chained)
	assert.Equal(t, want, got)
}

func Test3(t *testing.T) {
	words := wordladder.NewWords([]string{"adijA", "Akq", "qZR"})
	problem := wordladder.NewProblem(words)
	chained, _ := wordladder.New().Solve(problem)
	assert.True(t, chained)
}

File: wordladder/wordladder_test.py
import unittest

import wordladder


class TestWordLadder(unittest.TestCase):
    def test1(self) -> None:
        words = ["apPle", "error", "suBway", "Zb"]
        problem = wordladder.Problem(words)
        want = wordladder.Answer(chained=False, last="r", first="s")
        got = wordladder.solve(problem)
        self.assertEqual(want, got)

    def test2(self) -> None:
        words = ["idIOh", "hiKoQA", "AbijoD", "djgeldi", "kjoial"]
        problem = wordladder.Problem(words)
        want = wordladder.Answer(chained=False, last="D", first="d")
        got = wordladder.solve(problem)
        self.assertEqual(want, got)

    def test3(self) -> None:
        words = ["adijA", "Akq", "qZR"]
        problem = wordladder.Problem(words)
        want = wordladder.Answer(chained=True)
        got = wordladder.solve(problem)
        self.assertEqual(want, got)


if __name__ == "__main__":
    unittest.main()
